### PR TITLE
[mds-db] [mds-compliance] [mds-policy-author] Break readGeographies into separate methods for reading Geography and GeographySummary objects.

### PR DESCRIPTION
**Pull request by avatarneil**
Just some code-hygiene improvements  on my PR from earlier today.

## PR Checklist

 - [x] simple searchable title - `[mds-db] Add PG env var`, `[config] Fix eslint config`
 - [x] briefly describe the changes in this PR
 - [ ] mark as draft if should not be merged
 - [ ] write tests for all new functionality

## Impacts
- [ ] Provider
- [ ] Agency
- [ ] Audit
- [ ] Policy
- [x] Compliance
- [ ] Daily
- [ ] Native
- [x] Policy Author

### CODE REVIEW

File: packages/mds-db/geographies.ts [hunk 1]
```diff
@@ -25,18 +25,14 @@ export async function readSingleGeography(geography_id: UUID): Promise<Geography
...
   // query on ids
   // return geographies
   try {
     const client = await getReadOnlyClient()
 
-    const cols =
-      params && params.summary
-        ? [...schema.TABLE_COLUMNS.geographies].filter(col => col !== schema.COLUMN.geography_json).join(',')
-        : '*'
+    const cols = '*'
```
Comment (drtyh2o) [inline packages/mds-db/geographies.ts]: This seems less useful now.

File: packages/mds-db/geographies.ts [hunk 2]
```diff
@@ -65,6 +61,46 @@ export async function readGeographies(params?: {
...
+    }
+
+    if (conditions.length) {
+      sql += ` WHERE ${conditions.join(' AND ')}`
+    }
+
+    const values = vals.values()
+    // TODO insufficiently general
+    // TODO add 'count'
+    const { rows } = await client.query(sql, values)
```
Comment (drtyh2o) [inline packages/mds-db/geographies.ts]: I am not sure if there would be performance concerns, but this method _could_ just delegate to `await readGeographies` and then map out the `geography_json`.

File: packages/mds-db/geographies.ts [hunk 3]
```diff
@@ -25,18 +25,14 @@ export async function readSingleGeography(geography_id: UUID): Promise<Geography
 
 export async function readGeographies(params?: {
   get_read_only?: boolean
-  summary?: boolean
-}): Promise<(Geography | GeographySummary)[]> {
+}): Promise<Geography[]> {
```
Comment (drtyh2o) [inline packages/mds-db/geographies.ts]: Can we tidy this up? This is the pattern I prefer for options/params:
```ts
interface ReadGeographiesParams {
  get_read_only: boolean
}

export async function readGeographies(params: Partial<ReadGeographiesParams> = {}): Promise<Geography[]> {
  const { get_read_only } = { get_read_only: false, ...params }
  ...
}
```

File: packages/mds-db/geographies.ts [hunk 4]
```diff
@@ -65,6 +61,46 @@ export async function readGeographies(params?: {
     throw err
   }
 }
+
+export async function readGeographiesSummary(params?: {
```
Comment (drtyh2o) [inline packages/mds-db/geographies.ts]: Minor nit: Should this be `readGeographySummaries`? `readGeographiesSummary` sounds like something that would return me a single summary object.

File: packages/mds-db/geographies.ts [hunk 5]
```diff
@@ -65,6 +61,46 @@ export async function readGeographies(params?: {
...
+  get_read_only?: boolean
+}): Promise<GeographySummary[]> {
+  // use params to filter
+  // query on ids
+  // return geographies
```
Comment (janedotx) [inline packages/mds-db/geographies.ts]: that's a misleading comment

File: packages/mds-db/geographies.ts (hunk 2 above)
Comment (janedotx) [reply packages/mds-db/geographies.ts]: I agree with @drtyh2o , this is a lot of needlessly duplicated code. I don't think there's going to be a lot of geographies so I wouldn't be too concerned about performance right now.

File: packages/mds-db/geographies.ts (hunk 5 above)
Comment (avatarneil) [reply packages/mds-db/geographies.ts]: I kind of hate code comments in general (especially on a method as trivial as this), so more than happy to remove :)

File: packages/mds-db/index.ts [hunk 6]
```diff
@@ -203,7 +203,7 @@ export = {
   readTrips,
   readTripList,
   readGeographies,
-  readGeographiesSummary,
+  readGeographiesSummary: readGeographiesSummaries,
```
Comment (drtyh2o) [inline packages/mds-db/index.ts]: Bad VS Code rename here ..

File: packages/mds-db/geographies.ts [hunk 7]
```diff
@@ -23,18 +24,11 @@ export async function readSingleGeography(geography_id: UUID): Promise<Geography
   }
 }
 
-export async function readGeographies(params?: {
-  get_read_only?: boolean
-}): Promise<Geography[]> {
-  // use params to filter
-  // query on ids
-  // return geographies
+export async function readGeographies(params: Partial<ReadGeographiesParams> = {}): Promise<Geography[]> {
```
Comment (drtyh2o) [inline packages/mds-db/geographies.ts]: Would still prefer to see:
```
const { get_read_only } = { get_read_only: false, ...params }
```
To a) establish explicit default value, and b) obviate the need to use `params && params.get_read_only` below.

Comment (avatarneil) [reply packages/mds-db/geographies.ts]: I already did that 🤔

**Pull request merged**